### DISCUSSION TITLE
What is the optimal amount of time between writes to the GameCube?

### DISCUSSION THREAD

**Discussion by andrew171717** [NicoHood/Nintendo#72] [Q&A]
I've been trying to find the optimal amount of time between writes to the GameCube with not great luck. I'd like my project to work with a majority of games. I'm aware that each game polls the controller at different speeds. 

After a ton of testing, I think I've found the optimal amount of time (or at least a multiple of said time) between writes for the GameCube Gameboy Player. I have finally got proper inputs with it that are not turbo.
Which is roughly 50 microseconds, which is not a lot of time. However, when trying to play Melee, this appears to be too fast (or not within the time frame it is requesting inputs)... Is it even possible to use this library to work with multiple games or should I just focus on getting the main game I intent to play with working well?

Is there anyway to setup an interrupt with this library to respond whenever the GameCube requests input instead of trying to guess? This would be so nice.

**Comment by Skuzee** [comment]
The `console.write()` function is blocking, and is waiting for the data line to change state. It then reads the command sent from the gamecube and responds (i.e. with controller origin, data, etc) So you shouldn't have to do anything other than read the controller and then call `console.write()` 
If I recall correctly it takes about 350 microseconds to read the command and transfer the data, so trying to call something every 50us doesn't seem right.
